### PR TITLE
Move fixture `platform_api_conn` to common place. 

### PR DESCRIPTION
**Pull request by yutongzhang-microsoft**
### Description of PR

The `platform_api_conn` fixture is utilized by scripts in both the `smatswitch` and `platform_tests` directories. To reduce cross-feature dependencies and improve code organization, I relocated the fixture to a shared common location, making it accessible without creating unnecessary interdependencies between these feature-specific directories.

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
The `platform_api_conn` fixture is utilized by scripts in both the `smatswitch` and `platform_tests` directories. To reduce cross-feature dependencies and improve code organization, I relocated the fixture to a shared common location, making it accessible without creating unnecessary interdependencies between these feature-specific directories.

#### How did you do it?
I relocated the fixture `platform_api_conn` to a shared common location

### CODE REVIEW

File: tests/platform_tests/api/test_chassis.py [hunk 1]
```diff
@@ -121,53 +122,61 @@ def compare_value_with_device_facts(self, duthost, key, value, case_sensitive=Tr
...
+                      ):
         duthost = duthosts[enum_rand_one_per_hwsku_hostname]
         name = chassis.get_name(platform_api_conn)
         pytest_assert(name is not None, "Unable to retrieve chassis name")
         pytest_assert(isinstance(name, STRING_TYPE), "Chassis name appears incorrect")
         self.compare_value_with_platform_facts(duthost, 'name', name)
 
-    def test_get_presence(self, duthosts, enum_rand_one_per_hwsku_hostname, localhost, platform_api_conn):
+    def test_get_presence(self, duthosts, enum_rand_one_per_hwsku_hostname, localhost, platform_api_conn # noqa F811
+                          ):
```
Comment (wangxin) [inline tests/platform_tests/api/test_chassis.py:134]: Why can't have these 2 lines in 1 line?

Comment (yutongzhang-microsoft) [reply tests/platform_tests/api/test_chassis.py]: Fixed all format issues.

**Review by wangxin — APPROVED** [review]

**Comment by anamehra** [comment]
This PR breaks api/test_sfp.py and api/test_thermal.py. Is there a PR to fix those?

**Comment by yutongzhang-microsoft** [comment]
> This PR breaks api/test_sfp.py and api/test_thermal.py. Is there a PR to fix those?

PR #15807 is for fixing

**Pull request merged**